compiler/rustc_codegen_llvm/src/gotoc/rvalue.rs

```diff
@@ -766,10 +766,17 @@ impl<'tcx> GotocCtx<'tcx> {
         // Insert a CBMC-time size check, roughly:
         //     <Ty> local_temp = nondet();
         //     assert(__CPROVER_OBJECT_SIZE(&local_temp) == vt_size);
-        let temp_var = self.gen_temp_variable(ty, Location::none()).to_expr();
+        let temp_var = self.gen_temp_variable(ty.clone(), Location::none()).to_expr();
         let decl = Stmt::decl(temp_var.clone(), None, Location::none());
-        let check = Expr::eq(Expr::object_size(temp_var.address_of()), vt_size.clone());
-        let assert_msg = format!("Correct CBMC vtable size for {:?}", operand_type.kind());
+        let cbmc_size = if ty.clone().is_empty() {
+            // CBMC errors on passing a pointer to void to __CPROVER_OBJECT_SIZE, so just pass 0
```

Comment:
Are we guaranteed this is correct? Or is this a temporary hack?

Reply:
Added an assert for the `Never` case and a link showing that we do know the size is 0. This way, will get a more detailed error if this fails again for a different type.